tests/drivers/st77xx/app.config.test

```diff
@@ -1,5 +1,5 @@
 # this file enables modules defined in Kconfig. Do not use this file for
 # application configuration. This is only needed during migration.
-CONFIG_MODULE_ST7735=y
+CONFIG_MODULE_ST77XX=y
```

Comment:
I think a Kconfig file is needed in the application directory to select the right module driver based on the type of board, like it is done in the Makefile.

```kconfig
config APPLICATION
    select MODULE_ST7789 if BOARD_with_st7789
    select MODULE_ST7796 if BOARD_with_st7796
    select MODULE_ST7745 if !BOARD_with_st7789 && !BOARD_with_st7796
```
(untested)

Reply:
Do you mean a specific board by `BOARD_with_st7789` and `BOARD_with_st7796`?

I'm not sure, but wouldn't it also work with the following?
```
config APPLICATION
    select MODULE_ST7789 if HAVE_ST7789
    select MODULE_ST7796 if HAVE_ST7796
    select MODULE_ST7735 if !HAVE_ST7789 && !HAVE_ST7796
```

Reply:
> I think a Kconfig file is needed in the application directory to select the right module driver based on the type of board, like it is done in the Makefile.

@aabadie I'm still a bit unsure about the right way to do Kconfig modeling. Do we agree that the following approach is the right one from the user's point of view?
```
(Top) → Drivers → Display Device Drivers → ST77xx display driver
[ ] ST7735 display
[*] ST7789 display
[ ] ST7796 display
```
That is, the user selects `Drivers -> Display Device Drivers -> ST77xx display driver` and the default variants is/are selected by the corresponding `HAVE_ST77*` features, e.g:
```
config MODULE_ST7789
    bool "ST7789 display"
    default y if HAVE_ST7789
    depends on MODULE_ST77XX
```

Reply:
> but wouldn't it also work with the following?

That seems better indeed. I don't know if that is good though. Maybe @MrKevinWeiss can tell

Reply:
The current state seems correct to me.

"I want a generic st77xx device", if a board has it declared than it will select it by default and if not, fallback to the default `st7735`.